### DISCUSSION TITLE
Real-Time Data Updates Issue in Shiny Dashboard with tea

### DISCUSSION THREAD

**Discussion by hyfzi** [insightsengineering/teal#1265] [Q&A]
1. When building a Shiny dashboard using teal, the running app.r does not display the latest data contents in real-time when the data is updated。    
2.I have consulted relevant resources, which suggest using functions like reactiveFileReader. However, when building a Shiny app with teal, teal provides encapsulated modules. How should I proceed to make modifications?    
3.I'm a beginner in R Shiny. Can anyone help me solve this issue?

**Comment by donyunardi** [comment]
I don't think real-time data updates is possible at the moment within teal framework.

The closest that I could think of is to create some event to re-trigger the data read using `teal_data_module()`, then run something like `invalidateLater()` combined with JS (i.e sendCustomMessage) to automatically run the event (i.e. click a button) on a time interval.

**Reply by hyfzi** [reply]
Yes, I think so too. A while ago, my colleague asked the person in charge, and their response was that real-time data updates cannot be implemented for the time being.

**Comment by chlebowa** [comment]
If this is still an issue, have you tried adding your data to the `teal` app with `teal_data_module` and using `reactiveFileReader` there?

**Reply by hyfzi** [reply]
Thank you for your response. I haven't had much time to learn teal recently, but when I have some free time, I'll try the methods you mentioned. Thank you!
